src/Explorer/Tabs/CloudShellTab/Data/CloudShellClient.tsx

```diff
@@ -0,0 +1,137 @@
+import { v4 as uuidv4 } from "uuid";
+import { configContext } from "../../../../ConfigContext";
+import { userContext } from "../../../../UserContext";
+import { armRequest } from "../../../../Utils/arm/request";
+import {
+  Authorization,
+  CloudShellProviderInfo,
+  CloudShellSettings,
+  ConnectTerminalResponse,
+  NetworkType,
+  OsType,
+  ProvisionConsoleResponse,
+  SessionType,
+  ShellType,
+} from "../Models/DataModels";
+import { getLocale } from "../Utils/CommonUtils";
+
+export const getUserSettings = async (): Promise<CloudShellSettings> => {
+  return await armRequest<CloudShellSettings>({
+    host: configContext.ARM_ENDPOINT,
+    path: `/providers/Microsoft.Portal/userSettings/cloudconsole`,
+    method: "GET",
+    apiVersion: "2023-02-01-preview",
+  });
+};
+
+export const putEphemeralUserSettings = async (
+  userSubscriptionId: string,
+  userRegion: string,
+  vNetSettings?: object,
+) => {
+  const ephemeralSettings: CloudShellSettings = {
+    properties: {
+      preferredOsType: OsType.Linux,
+      preferredShellType: ShellType.Bash,
+      preferredLocation: userRegion,
+      networkType:
+        !vNetSettings || Object.keys(vNetSettings).length === 0
+          ? NetworkType.Default
+          : vNetSettings
+          ? NetworkType.Isolated
+          : NetworkType.Default,
+      sessionType: SessionType.Ephemeral,
+      userSubscription: userSubscriptionId,
+      vnetSettings: vNetSettings ?? {},
+    },
+  };
+
+  return await armRequest({
+    host: configContext.ARM_ENDPOINT,
+    path: `/providers/Microsoft.Portal/userSettings/cloudconsole`,
+    method: "PUT",
+    apiVersion: "2023-02-01-preview",
+    body: ephemeralSettings,
+  });
+};
+
+export const verifyCloudShellProviderRegistration = async (subscriptionId: string): Promise<CloudShellProviderInfo> => {
+  return await armRequest({
+    host: configContext.ARM_ENDPOINT,
+    path: `/subscriptions/${subscriptionId}/providers/Microsoft.CloudShell`,
+    method: "GET",
+    apiVersion: "2022-12-01",
+  });
+};
+
+export const registerCloudShellProvider = async (subscriptionId: string) => {
+  return await armRequest({
+    host: configContext.ARM_ENDPOINT,
+    path: `/subscriptions/${subscriptionId}/providers/Microsoft.CloudShell/register`,
+    method: "POST",
+    apiVersion: "2022-12-01",
+  });
+};
+
+export const provisionConsole = async (consoleLocation: string): Promise<ProvisionConsoleResponse> => {
+  const data = {
+    properties: {
+      osType: OsType.Linux,
+    },
+  };
+
+  return await armRequest<ProvisionConsoleResponse>({
+    host: configContext.ARM_ENDPOINT,
+    path: `providers/Microsoft.Portal/consoles/default`,
```

Comment:
ref. API https://learn.microsoft.com/en-us/azure/templates/microsoft.portal/consoles?pivots=deployment-language-arm-template

Reply:
Can you please add a comment with this reference ?